apps/builder/app/builder/features/style-panel/shared/use-intermediate-layer.ts

```diff
@@ -0,0 +1,46 @@
+import type { InvalidValue, LayersValue } from "@webstudio-is/css-engine";
+import { useState } from "react";
+
+type IntermediateValue = {
+  type: "intermediate";
+  value: string;
+};
+
+export const useIntermediateContent = ({
```

Comment:
First this hook is written with bug. It copypasted from buggy version. And this bug are in few places.
intermediate value should be used only during edit. And should be reseted after onComplete. Otherwise any changes to style from other places (like AI changes) will never be shown.

2nd I don't think that this hook need to be shared. It has no value to be shareable. I don't even like its a hook, as its just handlers and state.